### ISSUE TITLE
[Panel] The ms-Layer created by the panel will not be removed when the panel is closed

### ISSUE THREAD

**Issue by sam961124** [microsoft/fluentui#17462]
### Environment Information

- **Package version(s)**: 8.1.0
- **Browser and OS versions**: Chrome, Windows 10

### Description
Opening a Panel component will create a `<div className="ms-Layer">`  under `<body>`. If we close the panel, the `<div>` will not be removed. Hence, if we open and close the panel several times, we will get several divs under `<body>`. These divs seem useless and quite annoying.
This behavior is **inconsistent** with `office-ui-fabric-react`.
I have tried it with exact same code and it will not create extra div under  `<body>`.

### Repro steps:
Code
```
<button
    title="Help"
    onClick={() => this.setState({ isOpen: true })}
>
    <FontIcon iconName="Help" className="icon" />
</button>
<Panel
    isLightDismiss
    isOpen={this.state.isOpen}
    onDismiss={() => this.setState({ isOpen: false})}
    headerText="Help"
    closeButtonAriaLabel="Close"
></Panel>
```
- Click button to set `isOpen` of the panel to `true` to open it.
- Close the panel by triggering `onDismiss`.
- Repeat these two steps several times.

#### Actual behavior:
![image](https://user-images.githubusercontent.com/10041262/111427805-9ea9af80-8731-11eb-8b9e-26b5fbc3cf28.png)

#### Expected behavior:
Only see single `<div className="ms-Layer">`  when the panel is open.
If the panel is closed, no extra div will be presented under `<body>`.

**Comment by richardkooiman** [comment]
Hi @sam961124 and @khmakoto,

Fiddled with some codepens to try reproduce this issue but haven't managed so far (both a good thing but also not very helpful regarding this issue probably).

react 17.0.2
\+ office-ui-fabric-react 7.165.1 --> https://codesandbox.io/s/fluentui-panel-layer-office-ui-fabric-react-5f2z2
\+ @fluentui/react 8.5.1 (hooks) --> https://codesandbox.io/s/fluentui-panel-layer-fluentui-react-hooks-12udj
\+ @fluentui/react 8.5.1 (class components) --> https://codesandbox.io/s/fluentui-panel-layer-fluentui-react-class-components-5yxrf

So far I see no strange behaviour and all the layers are closing as expected. However, of course @sam961124's application calling the Panel component is probably a whole lot different than the codepens. Perhaps it's possible to try adjust the codepen a bit so it reflects the interaction with the Panel component as the application does a bit more?

**Comment by sam961124** [comment]
@richardkooiman Thanks for your works. I have tried them and did not see any strange behavior, too. 
In addition, I tried the codepens in @fluentui/react 8.1.0 (My app version), and I saw the strange behavior. I think this issue has been fixed in the latest version. I also update my package in the app to the latest version, and everything works fine. I will close the issue here. Thanks for your help again!
